services/subscriptions/subscriptions_test.go

```diff
@@ -0,0 +1,225 @@
+package subscriptions
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/status-im/status-go/signal"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	filterID = "123"
+	filterNS = "tst"
+)
+
+type mockFilter struct {
+	filterID       string
+	data           []interface{}
+	filterError    error
+	uninstalled    bool
+	uninstallError error
+}
+
+func newMockFilter(filterID string) *mockFilter {
+	return &mockFilter{
+		filterID: filterID,
+	}
+}
+
+func (mf *mockFilter) getID() string {
+	return mf.filterID
+}
+func (mf *mockFilter) getChanges() ([]interface{}, error) {
+	if mf.filterError != nil {
+		err := mf.filterError
+		mf.filterError = nil
+		return nil, err
+	}
+
+	data := mf.data
+	mf.data = nil
+	return data, nil
+}
+
+func (mf *mockFilter) uninstall() error {
+	mf.uninstalled = true
+	return mf.uninstallError
+}
+
+func (mf *mockFilter) setData(data ...interface{}) {
+	mf.data = data
+}
+
+func (mf *mockFilter) setError(err error) {
+	mf.data = nil
+	mf.filterError = err
+}
+
+func TestSubscriptionGetData(t *testing.T) {
+	filter := newMockFilter(filterID)
+
+	subs := NewSubscriptions(time.Microsecond)
+
+	subID, _ := subs.Create(filterNS, filter)
+
+	require.Equal(t, string(subID), fmt.Sprintf("%s-%s", filterNS, filterID))
+
+	proceed := make(chan struct{})
+
+	signal.SetDefaultNodeNotificationHandler(func(jsonEvent string) {
+		defer close(proceed)
+		validateFilterData(t, jsonEvent, string(subID), "1", "2", "3", "4")
+	})
+
+	filter.setData("1", "2", "3", "4")
+
+	select {
+	case <-proceed:
+		return
+	case <-time.After(time.Second):
+		require.NoError(t, errors.New("timeout while waiting for filter results"))
+	}
+
+	require.NoError(t, subs.RemoveAll())
+	signal.ResetDefaultNodeNotificationHandler()
+}
+
+func TestSubscriptionGetError(t *testing.T) {
+	filter := newMockFilter(filterID)
+
+	subs := NewSubscriptions(time.Microsecond)
+
+	subID, _ := subs.Create(filterNS, filter)
+
+	require.Equal(t, string(subID), fmt.Sprintf("%s-%s", filterNS, filterID))
+
+	proceed := make(chan struct{})
+
+	expectedError := errors.New("test-error")
+
+	signal.SetDefaultNodeNotificationHandler(func(jsonEvent string) {
+		defer close(proceed)
+		validateFilterError(t, jsonEvent, string(subID), expectedError.Error())
+	})
+
+	filter.setError(expectedError)
+
+	select {
+	case <-proceed:
+		return
+	case <-time.After(time.Second):
+		require.NoError(t, errors.New("timeout while waiting for filter results"))
+	}
+
+	require.NoError(t, subs.RemoveAll())
+	signal.ResetDefaultNodeNotificationHandler()
+}
+
+func TestSubscriptionRemove(t *testing.T) {
+	filter := newMockFilter(filterID)
+
+	subs := NewSubscriptions(time.Microsecond)
+
+	subID, _ := subs.Create(filterNS, filter)
+
+	require.NoError(t, subs.Remove(subID))
+
+	require.True(t, filter.uninstalled)
+	require.Equal(t, len(subs.subs), 0)
```

Comment:
Isn't `Empty` better?
```suggestion
	require.Empty(t, subs.subs)
```